### PR TITLE
add modal component

### PR DESCRIPTION
**Pull request by sk-ip**
review modal component

### CODE REVIEW

File: component/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+/* Typography */
+
+:root {
+    --font-family: system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue",
+    Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji",
+    "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+}
+
```
Comment (nyyshaaa) [inline component/modal/modal.css]: specify this in a utility file,as you are using it in every component.

**Review by nyyshaaa — APPROVED** [review]

File: component/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  position: fixed;
+  z-index: 1;
+  padding-top: 100px;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+  background-color: rgb(0,0,0); /* Fallback color */
+  background-color: rgba(0,0,0,0.4); /* Black w/ opacity */
```
Comment (Aditi0505) [inline component/modal/modal.css]: Can use hex values for consistency.

File: component/modal/modal.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+function ModalModule() {
+    const modalClose = document.querySelectorAll('.modal-close')
+
+    function hideModal(e) {
+        console.log(e)
```
Comment (Aditi0505) [inline component/modal/modal.js]: can remove console.log()

File: component/modal/modal.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        e.preventDefault();
+        e.target.parentElement.parentElement.style.visibility = 'hidden';
```
Comment (Aditi0505) [inline component/modal/modal.js:6]: Any specific reason to use visibility hidden instead of display none? Let me know your thoughts about this.

**Review by Aditi0505 — APPROVED** [review]
Left few suggestions. Rest looks good.

File: component/modal/modal.css (hunk 1 above)
Comment (sk-ip) [reply component/modal/modal.css]: yes i think so, will do that.

File: component/modal/modal.js (hunk 4 above)
Comment (sk-ip) [reply component/modal/modal.js:6]: Never thought that we can also use display:none, thanks for that.

File: component/modal/modal.css (hunk 2 above)
Comment (sk-ip) [reply component/modal/modal.css]: ok got it.

**Pull request merged**